pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyAuthenticationTest.java

```diff
@@ -0,0 +1,269 @@
+
+		Set<String> providers = new HashSet<String>();
+		providers.add(BasicAuthenticationProvider.class.getName());
+		conf.setAuthenticationProviders(providers);
+
+		conf.setClusterName("test");
+		Set<String> proxyRoles = new HashSet<String>();
+		proxyRoles.add("proxy");
+		conf.setProxyRoles(proxyRoles);
+        conf.setAuthenticateOriginalAuthData(true);
+		super.init();
+
+		createAdminClient();
+		producerBaseSetup();
+	}
+
+	@Override
+	@AfterMethod
+	protected void cleanup() throws Exception {
+		super.internalCleanup();
+	}
+
+	@Test
+	void testAuthentication() throws Exception {
+		log.info("-- Starting {} test --", methodName);
+
+		// Step 1: Create Admin Client
+		createAdminClient();
+		final String proxyServiceUrl = "pulsar://localhost:" + servicePort;
+		// create a client which connects to proxy and pass authData
+		String namespaceName = "my-property/my-ns";
+		String topicName = "persistent://my-property/my-ns/my-topic1";
+		String subscriptionName = "my-subscriber-name";
+		// expires after 6 seconds
+		String clientAuthParams = "entityType:client,expiryTime:" + (System.currentTimeMillis() + 6 * 1000);
+		// expires after 3 seconds
+		String proxyAuthParams = "entityType:proxy,expiryTime:" + (System.currentTimeMillis() + 3 * 1000);
+
+		admin.namespaces().grantPermissionOnNamespace(namespaceName, "proxy",
+				Sets.newHashSet(AuthAction.consume, AuthAction.produce));
+		admin.namespaces().grantPermissionOnNamespace(namespaceName, "client",
+				Sets.newHashSet(AuthAction.consume, AuthAction.produce));
+
+		// Step 2: Try to use proxy Client as a normal Client - expect exception
+		ProxyConfiguration proxyConfig = new ProxyConfiguration();
+		proxyConfig.setAuthenticationEnabled(true);
+		proxyConfig.setServicePort(servicePort);
+		proxyConfig.setWebServicePort(webServicePort);
+		proxyConfig.setBrokerServiceURL("pulsar://localhost:" + BROKER_PORT);
+
+		proxyConfig.setBrokerClientAuthenticationPlugin(BasicAuthentication.class.getName());
+		proxyConfig.setBrokerClientAuthenticationParameters(proxyAuthParams);
+
+		Set<String> providers = new HashSet<>();
+		providers.add(BasicAuthenticationProvider.class.getName());
+		proxyConfig.setAuthenticationProviders(providers);
+		proxyConfig.setForwardAuthorizationCredentials(true);
+		ProxyService proxyService = new ProxyService(proxyConfig);
+
+		proxyService.start();
+
+		// Step 3: Pass correct client params
+		PulsarClient proxyClient = createPulsarClient(proxyServiceUrl, clientAuthParams, 1);
+		proxyClient.newProducer().topic(topicName).create();
+		// Sleep for 4 seconds - wait for proxy auth params to expire
+		Thread.sleep(4 * 1000);
+		proxyClient.newProducer().topic(topicName).create();
+		// Sleep for 3 seconds - wait for client auth parans to expire
+		Thread.sleep(3 * 1000);
+		proxyClient.newProducer().topic(topicName).create();
+		proxyClient.close();
+		proxyService.close();
+	}
+
+	private void createAdminClient() throws PulsarClientException {
```

Comment:
`createAdminClient()` name seems misleading as we are not closing admin and we may think it's leaking. so, probably we can rename it as `updateAdmin()`?